image/docker_schema1.go

```diff
@@ -107,46 +107,53 @@ func (m *manifestSchema1) UpdatedImageNeedsLayerDiffIDs(options types.ManifestUp
 // This does not change the state of the original Image object.
 func (m *manifestSchema1) UpdatedImage(ctx context.Context, options types.ManifestUpdateOptions) (types.Image, error) {
 	copy := manifestSchema1{m: manifest.Schema1Clone(m.m)}
-	if options.LayerInfos != nil {
-		if err := copy.m.UpdateLayerInfos(options.LayerInfos); err != nil {
-			return nil, err
-		}
-	}
-	if options.EmbeddedDockerReference != nil {
-		copy.m.Name = reference.Path(options.EmbeddedDockerReference)
-		if tagged, isTagged := options.EmbeddedDockerReference.(reference.NamedTagged); isTagged {
-			copy.m.Tag = tagged.Tag()
-		} else {
-			copy.m.Tag = ""
-		}
-	}
 
 	switch options.ManifestMIMEType {
-	case "": // No conversion, OK
-	case manifest.DockerV2Schema1MediaType, manifest.DockerV2Schema1SignedMediaType:
+	// "" - No conversion
 	// We have 2 MIME types for schema 1, which are basically equivalent (even the un-"Signed" MIME type will be rejected if there isn’t a signature; so,
 	// handle conversions between them by doing nothing.
+	case "", manifest.DockerV2Schema1MediaType, manifest.DockerV2Schema1SignedMediaType:
+		if options.LayerInfos != nil {
+			if err := copy.m.UpdateLayerInfos(options.LayerInfos); err != nil {
+				return nil, err
+			}
+		}
+		if options.EmbeddedDockerReference != nil {
+			copy.m.Name = reference.Path(options.EmbeddedDockerReference)
+			if tagged, isTagged := options.EmbeddedDockerReference.(reference.NamedTagged); isTagged {
+				copy.m.Tag = tagged.Tag()
+			} else {
+				copy.m.Tag = ""
+			}
+		}
+
+		return memoryImageFromManifest(&copy), nil
 	case manifest.DockerV2Schema2MediaType:
 		m2, err := copy.convertToManifestSchema2(options.InformationOnly.LayerInfos, options.InformationOnly.LayerDiffIDs)
 		if err != nil {
 			return nil, err
 		}
-		return memoryImageFromManifest(m2), nil
+
+		return m2.UpdatedImage(ctx, options)
```

Comment:
And it would be nice to have the “no conversion” edits (`LayerInfos`+`EmbeddedDockerReference`) visually/structurally separated from the “convert depending on MIME type” path. Doing that without duplicating the “no conversion needs to happen” check rather calls for a helper to do the conversion + `UpdatedImage` call (and that helper could drop the `ManifestMIMEType` edit request so that unexpected no-ops are detected, see elsewhere). 

Maybe
```go
converted, err := convertManifestIfRequired(options, map[string]func(context.Context, types.ManifestUpdateInformation)  (types.Image, error) {
    manifest.DockerV2Schema2MediaType: copy.convertToManifestSchema2,
    imgspecv1.MediaTypeImageManifest:  copy.convertToOCI, // a new helper
})
if err != nil { fail }
if converted != nil { return converted }
// ordinary edits for LayerInfos and EmbeddedDocerReference follow
```
with a
```go
func convertManifestIfRequired(options types.ManifestUpdateOptions, converters map[string]func(context.Context, types.ManifestUpdateInformation)  (types.Image, error) {
    if options.ManifestMIMEType != "" {
        if converter, ok := converters[options.ManifestMIMEType]; ok {
            tmp, err := converter(options.InformationOnly)
            if err != nil { fail }
            optionsCopy := options
            optionsCopy.ManifestMIMEType = ""
            return tmp.UpdatedImage(ctx, optionsCopy)
        }
    }
    return nil, nil
}
```

Reply:
Yea this makes sense. Let me make this change.